pkg/timeutils/utils.go

```diff
@@ -5,15 +5,23 @@ import (
 	"time"
 )
 
-// GetTimestamp tries to parse given string as RFC3339 time
-// or Unix timestamp, if successful returns a Unix timestamp
-// as string otherwise returns value back.
-func GetTimestamp(value string) string {
+// GetTimestamp tries to parse given string as golang duration,
+// then RFC3339Nano time and finally as Unix timestamp. If
+// any of these were successful, it returns a Unix timestamp
+// as string otherwise returns the given value back.
+// In case of duration input, the returned timestamp is computed
+// as the given reference time minus the amount of the duration.
+func GetTimestamp(value string, reference time.Time) string {
+	if d, err := time.ParseDuration(value); value != "0" && err == nil {
```

Comment:
I don't know but I think I'm more comfortable in splitting this in d, err := ... and the condition below (although we don't care about d if err != nil), more readable maybe, fell free to skip this comment btw, just my brain